runtime/src/main/java/org/corfudb/runtime/RoutingQueueSenderClient.java

```diff
@@ -0,0 +1,533 @@
+     * @param clientName String representation of the client name. This parameter is case-sensitive.
+     * @throws IllegalArgumentException If clientName is null or empty.
+     * @throws NoSuchMethodException    NoSuchMethodException.
+     * @throws IllegalAccessException   IllegalAccessException.
+     */
+    public RoutingQueueSenderClient(CorfuStore corfuStore, String clientName)
+            throws NoSuchMethodException, IllegalAccessException {
+        Preconditions.checkArgument(isValid(clientName), "clientName is null or empty.");
+
+        this.corfuStore = corfuStore;
+        this.clientName = clientName;
+
+        int numOpenQueueRetries = 8;
+        Table<CorfuGuidMsg, RoutingTableEntryMsg, CorfuQueueMetadataMsg> logEntryQ_local = null;
+        Table<CorfuGuidMsg, RoutingTableEntryMsg, CorfuQueueMetadataMsg> snapSyncQ_local = null;
+        Table<RoutingQSnapSyncHeaderKeyMsg, RoutingQSnapSyncHeaderMsg, Message> snapStartEnd_local = null;
+        while ((numOpenQueueRetries--) > 0) {
+            try {
+                logEntryQ_local = corfuStore.openQueue(CORFU_SYSTEM_NAMESPACE, LOG_ENTRY_SYNC_QUEUE_NAME_SENDER,
+                        RoutingTableEntryMsg.class, TableOptions.fromProtoSchema(RoutingTableEntryMsg.class));
+                snapSyncQ_local = corfuStore.openQueue(CORFU_SYSTEM_NAMESPACE, SNAPSHOT_SYNC_QUEUE_NAME_SENDER,
+                        RoutingTableEntryMsg.class, TableOptions.fromProtoSchema(RoutingTableEntryMsg.class));
+                snapStartEnd_local = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, SNAP_SYNC_TXN_ENVELOPE_TABLE,
+                        RoutingQSnapSyncHeaderKeyMsg.class, RoutingQSnapSyncHeaderMsg.class, null,
+                        TableOptions.fromProtoSchema(RoutingTableEntryMsg.class));
+
+                break;
+            } catch (InvocationTargetException e) {
+                throw new RuntimeException("InvocationTargetException in fromProtoSchema" + e.getMessage());
+            } catch (TransactionAbortedException e) {
+                log.warn("OpenQueue in RoutingQSender hit TAE: retry" + numOpenQueueRetries);
+            } catch (StreamingException se) {
+                log.warn("RoutingQSender subscription hit a Streaming Exception retrying " + numOpenQueueRetries);
+            }
+        }
+
+        while ((numOpenQueueRetries--) > 0) {
+            try {
+                replicationEventTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
+                        ReplicationEventInfoKey.class,
+                        ReplicationEvent.class,
+                        null,
+                        TableOptions.fromProtoSchema(ReplicationEvent.class));
+                break;
+            } catch (InvocationTargetException | IllegalArgumentException e) {
+                log.error("Failed to open the Event Table", e);
+                throw new RuntimeException(e);
+            }
+        }
```

Comment:
Similar to [this](https://github.com/CorfuDB/CorfuDB/pull/3799#discussion_r1364322705) comment, are the retries here needed?

Reply:
I guess not, we can remove the retry from here.

Reply:
ok

Reply:
Removed the retries and consolidated this logic